packages/ckeditor5-ui/src/search/text/searchtextview.ts

```diff
@@ -0,0 +1,388 @@
+	 */
+	public queryView: SearchTextQueryView<TQueryFieldView>;
+
+	/**
+	 * Controls whether the component is in read-only mode.
+	 *
+	 * @default true
+	 */
+	declare public isEnabled: boolean;
+
+	/**
+	 * The number of results found for the current search query. Updated upon the {@link #search} event.
+	 *
+	 * @default 0
+	 */
+	declare public resultsCount: number;
+
+	/**
+	 * The number of the items that can be searched in the {@link #filteredView}. Updated upon the {@link #search} event.
+	 *
+	 * @default 0
+	 */
+	declare public totalItemsCount: number;
+
+	/**
+	 * The collection of children of the view.
+	 *
+	 * @readonly
+	 */
+	declare public readonly children: ViewCollection;
+
+	/**
+	 * The collection of focusable children of the view. Used by the focus management logic.
+	 *
+	 * @readonly
+	 */
+	declare public readonly focusableChildren: ViewCollection;
+
+	/**
+	 * Provides the focus management (keyboard navigation) between {@link #queryView} and {@link #filteredView}.
+	 *
+	 * @readonly
+	 */
+	private _focusCycler: FocusCycler;
+
+	/**
+	 * The cached configuration object.
+	 *
+	 * @internal
+	 */
+	protected _config: SearchTextViewConfig<TQueryFieldView>;
+
+	/**
+	 * Creates an instance of the {@link module:ui/search/text/searchtextview~SearchTextView} class.
+	 *
+	 * @param locale The localization services instance.
+	 * @param config Configuration of the view.
+	 */
+	constructor( locale: Locale, config: SearchTextViewConfig<TQueryFieldView> ) {
+		super( locale );
+
+		this._config = config;
+
+		this.filteredView = config.filteredView;
+		this.queryView = this._createSearchTextQueryView();
+		this.focusTracker = new FocusTracker();
+		this.keystrokes = new KeystrokeHandler();
+		this.resultsView = new SearchResultsView( locale );
+		this.children = this.createCollection();
+		this.focusableChildren = this.createCollection( [ this.queryView, this.resultsView ] );
+
+		this.set( 'isEnabled', true );
+		this.set( 'resultsCount', 0 );
+		this.set( 'totalItemsCount', 0 );
+
+		if ( config.infoView && config.infoView.instance ) {
+			this.infoView = config.infoView.instance;
+		} else {
+			this.infoView = new SearchInfoView();
+			this._enableDefaultInfoViewBehavior();
+
+			this.on( 'render', () => {
+				// Initial search that determines if there are any searchable items
+				// and displays the corresponding info text.
+				this.search( '' );
+			} );
+		}
+
+		this.resultsView.children.addMany( [ this.infoView, this.filteredView ] );
+
+		this._focusCycler = new FocusCycler( {
+			focusables: this.focusableChildren,
+			focusTracker: this.focusTracker,
+			keystrokeHandler: this.keystrokes,
+			actions: {
+				// Navigate form fields backwards using the Shift + Tab keystroke.
+				focusPrevious: 'shift + tab',
+
+				// Navigate form fields forwards using the Tab key.
+				focusNext: 'tab'
+			}
+		} );
+
+		this.on( 'search', ( evt, { resultsCount, totalItemsCount } ) => {
```

Comment:
```suggestion
		this.on<SearchTextViewSearchEvent>( 'search', ( evt, { resultsCount, totalItemsCount } ) => {
```
Currently, `resultsCount` and `totalItemsCount` are `any`.